src/main/java/com/azure/cosmos/examples/crudquickstart/async/SampleCRUDQuickstartAsync.java

```diff
@@ -0,0 +1,342 @@
+
+        try {
+            completionLatch.await();
+        } catch (InterruptedException err) {
+            throw new AssertionError("Unexpected Interruption",err);
+        }
+
+        //  </CreateItem>
+    }
+
+    private void upsertFamily(Family family_to_upsert) {
+        //Modify a field of the family object
+        System.out.println(String.format("Upserting the item with id %s after modifying the isRegistered field...",family_to_upsert.getId()));
+        family_to_upsert.setRegistered(!family_to_upsert.isRegistered());
+
+        //Upsert the modified item
+        Mono.just(family_to_upsert).flatMap(item -> {
+            CosmosAsyncItemResponse<Family> item_resp = container.upsertItem(family_to_upsert).block();
+
+            //  Get upsert request charge and other properties like latency, and diagnostics strings, etc.
+            System.out.println(String.format("Upserted item with request charge of %.2f within duration %s",
+                    item_resp.getRequestCharge(), item_resp.getRequestLatency()));
+
+            return Mono.empty();
+        }).subscribe();
+    }
+
+    private void readItems(Flux<Family> familiesToCreate) {
+        //  Using partition key for point read scenarios.
+        //  This will help fast look up of items because of partition key
+        //  <ReadItem>
+
+        final CountDownLatch completionLatch = new CountDownLatch(1);
+
+        familiesToCreate.flatMap(family -> {
+                            Mono<CosmosAsyncItemResponse<Family>> asyncItemResponseMono = container.readItem(family.getId(), new PartitionKey(family.getLastName()), Family.class);
+                            return asyncItemResponseMono;
+                        })
+                        .subscribe(
+                            itemResponse -> {
+                                double requestCharge = itemResponse.getRequestCharge();
+                                Duration requestLatency = itemResponse.getRequestLatency();
+                                System.out.println(String.format("Item successfully read with id %s with a charge of %.2f and within duration %s",
+                                    itemResponse.getResource().getId(), requestCharge, requestLatency));
+                            },
+                            err -> {
+                                if (err instanceof CosmosClientException) {
+                                    //Client-specific errors
+                                    CosmosClientException cerr = (CosmosClientException)err;
+                                    cerr.printStackTrace();
+                                    System.err.println(String.format("Read Item failed with %s\n", cerr));
+                                } else {
+                                    //General errors
+                                    err.printStackTrace();
+                                }
+
+                                completionLatch.countDown();
```

Comment:
we should try to remove countdown latch for samples, they are hard to understand.